### PR TITLE
Return resulting ChangesetEvent from Changeset.Merge

### PR DESCRIPTION
**Pull request by renfredxh**

### CODE REVIEW

File: sourcegraph/sourcegraph.proto [hunk 1]
```diff
@@ -809,8 +809,9 @@ service Changesets {
 	rpc Update(ChangesetUpdateOp) returns (ChangesetEvent);
 
 	// Merge merges the head branch of a changeset into its base branch and
-	// pushes the resulting merged base.
-	rpc Merge(ChangesetMergeOp) returns (pbtypes.Void);
+	// pushes the resulting merged base. It returns the resulting update event.
+	// If no merge occurred, it returns nil.
```
Comment (shurcooL) [inline sourcegraph/sourcegraph.proto]: > If no merge occurred, it returns nil.

That sounds great, but is it allowed by gRPC?

Comment (renfredxh) [inline sourcegraph/sourcegraph.proto]: I believe so, for reference the same thing happens in [Changeset.Update](https://github.com/sourcegraph/go-sourcegraph/blob/changeset-merge-event/sourcegraph/sourcegraph.proto#L807-L809)

Comment (shurcooL) [inline sourcegraph/sourcegraph.proto]: Good to know, thanks for confirming.

I think it's just the input parameters that are not allowed to be `nil` then.

**Comment by shurcooL** [comment]
LGTM.

**Pull request merged**